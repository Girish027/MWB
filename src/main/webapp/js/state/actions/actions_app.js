
import _ from 'lodash';
import getUrl, { pathKey } from 'utils/apiUrls';
import { fetchUtil } from 'utils/fetchUtils';
import goToRoute from 'utils/routeHelpers';
import Constants from 'constants/Constants';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { logAmplitudeEvent } from 'utils/amplitudeUtils';
import axios from 'axios';
import api from 'utils/api';
import { setDefaultsForAPI } from 'utils/api';
import * as types from './types';

// Note to developers:
// Both Warning and Good Messages are shown for the period of notificationInterval.
// Bad, Persistant - till user closes the notification. This should only be used sparingly.
export const callLogIngestAPI = (message, logLevel) => {
  const promise = new Promise((resolve, reject) => {
    const url = getUrl(pathKey.logIngest, { logLevel });
    return new Promise((resolve, reject) => {
      api.post({
        url,
        headers: {
          'Content-Type': 'text/plain',
        },
        data: message,
      });
    });
  });
};

export const displayBadRequestMessage = (message) => (dispatch) => {
  dispatch({
    type: types.BAD_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_ERROR);
};

export const displayWarningRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.WARNING_REQUEST,
    value: message,
  });

  callLogIngestAPI(message, Constants.LOG_LEVEL_WARNING);

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayGoodRequestMessage = (message, notificationInterval = Constants.NOTIFICATION.interval) => (dispatch) => {
  dispatch({
    type: types.GOOD_REQUEST,
    value: message,
  });

  setTimeout(() => {
    dispatch(stopShowingServerMessage());
  }, notificationInterval);
};

export const displayPersistantMessage = message => ({
  type: types.PERSISTENT_REQUEST,
  value: message,
});

export const stopShowingServerMessage = () => ({
  type: types.APP_STOP_SHOWING_SERVER_MESSAGE,
});

export const updateLoginSuccess = csrfToken => ({
  type: types.LOGIN_SUCCESS,
  value: csrfToken,
});

export const updateLoginInfo = (userId, userName, userDetails) => ({
  type: types.LOGIN_INFO,
  value: {
    userId,
    userName,
    userDetails,
  },
});

export const logoutFromOktaSuccess = () => ({
  type: types.LOGOUT_OKTA,
});

export const logoutFromMWBSuccess = () => ({
  type: types.LOGOUT_MWB,
});

export const contextMenuChange = contextMenuState => ({
  type: types.CONTEXT_MENU_CHANGE,
  contextMenuState,
});

export const modalDialogChange = modalDialogState => ({
  type: types.MODAL_DIALOG_CHANGE,
  modalDialogState,
});

const actionFileRemoveRequest = ({ fileId }) => ({
  type: types.FILE_REMOVE_REQUEST,
  fileId,
});

const actionFileRemoveError = ({ fileId, error }) => ({
  type: types.FILE_REMOVE_ERROR,
  fileId,
  error,
});

const actionFileRemoveSuccess = ({ fileId }) => ({
  type: types.FILE_REMOVE_SUCCESS,
  fileId,
});

export const updateUserGroups = data => ({
  type: types.UPDATE_USER_GROUPS,
  data,
});

const onFetchFailureRemoveFile = (error, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveError({ fileId, error }));
};

const onFetchSuccessRemoveFile = (data, params) => (dispatch) => {
  const { fileId } = params;
  dispatch(actionFileRemoveSuccess({ fileId }));
};

export const removeFile = ({ fileId }) => (dispatch, getState) => {
  dispatch(actionFileRemoveRequest({ fileId }));

  const state = getState();
  const fetchUrl = getUrl(pathKey.files, { fileId, clientId: state.header.client.id });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'delete',
    onFetchSuccess: onFetchSuccessRemoveFile,
    onFetchFailure: onFetchFailureRemoveFile,
    dispatch,
    getState,
    params: {
      fileId,
    },
  });
};

const onFetchSuccessGetUserGroups = data => (dispatch) => {
  dispatch(updateUserGroups(data));
};

export const getUserGroups = ({ csrfToken }) => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.userGroups, { csrfToken });

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetUserGroups,
    dispatch,
    getState,
  });
};

export const onFetchSuccessGetCSRFToken = data => (dispatch, getState) => {
  const csrfToken = data.value;
  setDefaultsForAPI(csrfToken);
  dispatch(updateLoginSuccess(csrfToken));
  dispatch(getUserGroups({ csrfToken }));
};

export const getCSRFToken = userId => (dispatch, getState) => {
  const fetchUrl = getUrl(pathKey.csrfToken);

  fetchUtil({
    fetchUrl,
    fetchMethod: 'get',
    onFetchSuccess: onFetchSuccessGetCSRFToken,
    dispatch,
    getState,
    params: {
      userId,
    },
  });
};

const logoutFromMWB = () => (dispatch, getState) => {
  dispatch(logoutFromOktaSuccess());
  dispatch(logoutFromMWBSuccess());
};

export const logoutFromOkta = () => (dispatch, getState) => {
  const state = getState();
  const { oAuthLogoutURL } = state.app;
  if (oAuthLogoutURL) {
    fetch(oAuthLogoutURL, {
      method: 'DELETE',
      credentials: 'include',
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    })
      .then((response) => {
        window.location.href = Constants.LOGOUT;
        switch (response.status) {
        case 204:
          logoutFromMWB();
          logAmplitudeEvent(AmplitudeConstants.LOGOUT_EVENT, state);
          break;
        case 404:
        default:
          // TODO send log for these scenarios
          break;
        }
      })
      .catch((error) => {
        // TODO send log for these scenarios
      });
  }
};

export const changeRoute = (routeName, data, history) => (dispatch, getState) => {
  const state = getState();
  const routeData = Object.assign({}, {
    client: !_.isNil(state.header.client) ? state.header.client : {
      id: 0,
      name: 'Loading...',
    },
    projectId: state.selectedProjectId,
  }, data);
  goToRoute(routeName, routeData, history);
};
