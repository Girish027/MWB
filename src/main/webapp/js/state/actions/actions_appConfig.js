import 'regenerator-runtime/runtime';

import getUrl, { pathKey } from 'utils/apiUrls';

import { fetchUtil } from 'utils/fetchUtils';
import AmplitudeConstants from 'constants/AmplitudeConstants';
import { initializeAnalytics, logAmplitudeEvent } from 'utils/amplitudeUtils';
import api from 'utils/api';
import * as types from './types';

const appConfigSuccess = data => ({
  type: types.APP_CONFIG_SUCCESS,
  data,
});

const appConfigFailure = data => ({
  type: types.APP_CONFIG_FAILURE,
  data,
});

const appConfigRequest = () => ({
  type: types.APP_CONFIG_REQUEST,
});

export const onAppConfigSuccess = data => (dispatch, getState) => {
  dispatch(appConfigSuccess(data));
  const state = getState();
  initializeAnalytics(state);
  logAmplitudeEvent(AmplitudeConstants.LOGIN_EVENT, state);
};

export const onAppConfigFailure = error => (dispatch) => {
  dispatch(appConfigFailure(error));
};

export const fetchAppConfig = () => (dispatch, getState) => {
  dispatch(appConfigRequest());
  const fetchUrl = getUrl(pathKey.appConfig);

  api.get({
    url: fetchUrl,
    onApiSuccess: onAppConfigSuccess,
    onApiError: onAppConfigFailure,
    dispatch,
    getState,
  });
};
