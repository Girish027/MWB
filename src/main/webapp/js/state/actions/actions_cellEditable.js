import * as types from './types';

export const stateCreate = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_CREATE,
  stateKey,
});

export const stateReset = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_RESET,
  stateKey,
});

export const stateRemove = ({ stateKey }) => ({
  type: types.CELL_EDITABLE_STATE_REMOVE,
  stateKey,
});

export const setActiveCell = ({ stateKey, activeRowIndex, activeColumnIndex }) => ({
  type: types.CELL_EDITABLE_SET_ACTIVE_CELL,
  stateKey,
  activeRowIndex,
  activeColumnIndex,
});

export const updateValue = ({ stateKey, value }) => ({
  type: types.CELL_EDITABLE_UPDATE_VALUE,
  stateKey,
  value,
});
