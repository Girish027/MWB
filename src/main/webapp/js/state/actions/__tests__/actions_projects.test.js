jest.mock('utils/api');

import api from 'utils/api';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import getUrl, { pathKey } from 'utils/apiUrls';
import * as actionsProjects from 'state/actions/actions_projects';

const middlewares = [thunk];
const reduxMockStore = configureMockStore(middlewares);


describe('actions_projects', () => {
  let store;
  let data;
  let clientId = '123';
  let mockPush;
  let testHistory;
  const startIndex = 0;
  const limit = 100;
  let projects = [];
  let projectId = '123';

  beforeAll(() => {
    store = reduxMockStore({
      app: {
        userId: 'testUser',
      },
      header: {
        client: {
          id: '14',
        },
      },
      projectListSidebar: {
        projectById: {
          123: {
            name: 'OldName',
          },
        },
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
    store.clearActions();
  });

  describe('createProject:', () => {
    beforeAll(() => {
      data = {
        values: {
          name: 'TestProject',
          vertical: 'FINANCIAL',
          description: 'TestProjectUpdated',
          locale: 'en-US',
          inheritTags: 'djkskd',
        },
        clientId: '14',
      };
      mockPush = jest.fn();
      testHistory = {
        push: mockPush,
      };
      clientId = '14';
    });

    test('should do post call to create project with required data', () => {
      store.dispatch(actionsProjects.createProject(data));
      expect(api.post).toHaveBeenCalledWith({
        dispatch: expect.any(Function),
        getState: expect.any(Function),
        url: expect.any(String),
        data: expect.any(Object),
        onApiSuccess: expect.any(Function),
        onApiError: actionsProjects.onCreateProjectFailure,
      });
    });

    test('should do post call with required data', () => {
      store.dispatch(actionsProjects.createProject(data));
      const url = api.post.mock.calls[0][0].url;
      expect(url).toMatchSnapshot();
    });
  });

  describe('onCreateProjectSuccess:', () => {
    let response = {
      name: 'TestProject',
    };

    test('should dispatch actions to create project and display good request message on project create success', () => {
      store.dispatch(actionsProjects.onCreateProjectSuccess(response, clientId, testHistory));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should dispatch actions to create project and display good request message on project create success without data', () => {
      store.dispatch(actionsProjects.onCreateProjectSuccess(undefined, clientId, testHistory));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('onCreateProjectFailure:', () => {
    let error = {
      message: 'Unable to create project',
    };

    let errorMsg = {
      ...error, code: 409,
    };
    test('should dispatch actions to display bad request message when project creation failed', () => {
      store.dispatch(actionsProjects.onCreateProjectFailure(error));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should dispatch actions to display warning message when project creation failed', () => {
      store.dispatch(actionsProjects.onCreateProjectFailure(errorMsg));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should dispatch actions to display warning message when project creation failed without data', () => {
      store.dispatch(actionsProjects.onCreateProjectFailure());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('deleteProjectById:', () => {
    test('should do delete call to delete project with required data', () => {
      store.dispatch(actionsProjects.deleteProjectById(projectId));
      expect(api.delete).toHaveBeenCalledWith({
        dispatch: expect.any(Function),
        getState: expect.any(Function),
        url: expect.any(String),
        onApiSuccess: expect.any(Function),
        onApiError: expect.any(Function),
      });
    });

    test('should do delete call with required data', () => {
      store.dispatch(actionsProjects.deleteProjectById(projectId));
      const url = api.delete.mock.calls[0][0].url;
      expect(url).toMatchSnapshot();
    });
  });

  describe('onDeleteProjectSuccess:', () => {
    test('should dispatch actions to delete project and display good request message on project delete success', () => {
      store.dispatch(actionsProjects.onDeleteProjectSuccess(projectId));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('updateByProject:', () => {
    beforeAll(() => {
      data = {
        values: {
          name: 'TestProject',
          vertical: 'FINANCIAL',
          description: 'TestProjectUpdated',
          locale: 'en-US',
        },
        clientId: '14',
        projectId: '123',
      };
    });

    test('should do patch call to update project with required data', () => {
      store.dispatch(actionsProjects.updateByProject(data));
      expect(api.patch).toHaveBeenCalledWith({
        dispatch: expect.any(Function),
        getState: expect.any(Function),
        url: expect.any(String),
        headers: expect.any(Object),
        data: expect.any(Array),
        onApiSuccess: expect.any(Function),
        onApiError: expect.any(Function),
      });
    });

    test('should do patch call to update project without data', () => {
      const dataMock = {
        values: {},
        clientId: '14',
        projectId: '123',
      };
      store.dispatch(actionsProjects.updateByProject(dataMock));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should do post call with required data', () => {
      store.dispatch(actionsProjects.updateByProject(data));
      const url = api.patch.mock.calls[0][0].url;
      expect(url).toMatchSnapshot();
    });
  });

  describe('onUpdateProjectSuccess:', () => {
    let response = [{
      id: '123',
      name: 'TestProject',
    }];

    test('should dispatch actions to update project status and display good request message on project update success', () => {
      store.dispatch(actionsProjects.onUpdateProjectSuccess(response, projectId));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should dispatch actions to update project status and display good request message on project update success without response available', () => {
      store.dispatch(actionsProjects.onUpdateProjectSuccess(undefined, projectId));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('promoteProjectById:', () => {
    beforeAll(() => {
      data = {
        projectId: '234',
        globalProjectId: '123',
        globalProjectName: 'Root_Intent',
      };
    });

    test('should do put call to promote project with required data', () => {
      store.dispatch(actionsProjects.promoteProjectById(data));
      expect(api.put).toHaveBeenCalledWith({
        dispatch: expect.any(Function),
        getState: expect.any(Function),
        url: expect.any(String),
        onApiSuccess: expect.any(Function),
        onApiError: expect.any(Function),
      });
    });

    test('should do put call with required data', () => {
      store.dispatch(actionsProjects.promoteProjectById(data));
      const url = api.put.mock.calls[0][0].url;
      expect(url).toMatchSnapshot();
    });
  });

  describe('onPromoteProjectSuccess:', () => {
    let response = [{
      id: '123',
      name: 'TestProject',
      type: 'NODE',
    }, {
      id: '234',
      name: 'Root_Intent',
      type: 'GLOBAL',
    }];

    test('should dispatch actions to promote project to global model and demote already existing project to node level model', () => {
      store.dispatch(actionsProjects.onPromoteProjectSuccess(response, projectId));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should dispatch actions to promote project to global model and demote already existing project to node level model without data available', () => {
      store.dispatch(actionsProjects.onPromoteProjectSuccess(undefined, projectId));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('demoteProjectById:', () => {
    test('should do put call to demote project with projectId', () => {
      store.dispatch(actionsProjects.demoteProjectById(projectId));
      expect(api.put).toHaveBeenCalledWith({
        dispatch: expect.any(Function),
        getState: expect.any(Function),
        url: expect.any(String),
        onApiSuccess: expect.any(Function),
        onApiError: expect.any(Function),
      });
    });

    test('should do put call with projectId', () => {
      store.dispatch(actionsProjects.demoteProjectById(projectId));
      const url = api.put.mock.calls[0][0].url;
      expect(url).toMatchSnapshot();
    });
  });

  describe('onProjectFailure:', () => {
    let error = {
      message: 'Unable to perform project action',
    };
    let errorMsg = {
      code: 409,
      message: 'Unable to perform project action',
    };

    test('should dispatch actions to set failed status and display bad request message when project operation failed', () => {
      store.dispatch(actionsProjects.onProjectFailure(error, projectId));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should dispatch actions to set warning status and display bad request message when project operation failed', () => {
      store.dispatch(actionsProjects.onProjectFailure(errorMsg, projectId));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should dispatch actions to set failed status and display bad request message when project operation failed without data available', () => {
      store.dispatch(actionsProjects.onProjectFailure(undefined, projectId));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('onMarkDeployableModelFailure:', () => {
    let error = {
      message: 'Unable to perform project action',
    };
    let modelVersion = '12';

    test('should dispatch actions to set failed status and display bad request message when mark for deploy operation failed', () => {
      store.dispatch(actionsProjects.onMarkDeployableModelFailure(error, modelVersion));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should dispatch actions to set failed status and display bad request message when mark for deploy operation failed without data', () => {
      store.dispatch(actionsProjects.onMarkDeployableModelFailure(undefined, modelVersion));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('markDeployableModel:', () => {
    beforeAll(() => {
      data = {
        clientId: '123',
        projectId: '456',
        modelId: '789',
      };
    });

    test('should do patch call to mark for deploy with required data', () => {
      store.dispatch(actionsProjects.markDeployableModel(data));

      const url = getUrl(pathKey.projectById, { clientId: '123', projectId: '456' });

      expect(api.patch).toHaveBeenCalledWith({
        dispatch: expect.any(Function),
        getState: expect.any(Function),
        url,
        headers: expect.any(Object),
        data: expect.any(Array),
        onApiSuccess: expect.any(Function),
        onApiError: expect.any(Function),
      });
    });
  });

  describe('unmarkDeployableModel:', () => {
    beforeAll(() => {
      data = {
        clientId: '123',
        projectId: '456',
        modelId: null,
      };
    });

    test('should do patch call to unmark for deploy', () => {
      store.dispatch(actionsProjects.unmarkDeployableModel(data));

      const url = getUrl(pathKey.projectById, { clientId: '123', projectId: '456' });

      expect(api.patch).toHaveBeenCalledWith({
        dispatch: expect.any(Function),
        getState: expect.any(Function),
        url,
        headers: expect.any(Object),
        data: expect.any(Array),
        onApiSuccess: expect.any(Function),
        onApiError: expect.any(Function),
      });
    });
  });

  describe('onUnmarkDeployableModelFailure:', () => {
    let error = {
      message: 'Unable to perform project action',
    };
    let modelVersion = '12';

    test('should dispatch actions to set failed status and display bad request message when unmark operation failed', () => {
      store.dispatch(actionsProjects.onUnmarkDeployableModelFailure(error, modelVersion));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should dispatch actions to set failed status and display bad request message when unmark operation failed without data', () => {
      store.dispatch(actionsProjects.onUnmarkDeployableModelFailure(undefined, modelVersion));
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('<Function call />', () => {
    test('should call requestProject', () => {
      store.dispatch(actionsProjects.requestProject('123'));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call toggleSidebarVisible', () => {
      store.dispatch(actionsProjects.toggleSidebarVisible(true));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call requestAllProjects', () => {
      store.dispatch(actionsProjects.requestAllProjects());
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call requestProjectsByClient', () => {
      store.dispatch(actionsProjects.requestProjectsByClient({ clientId, startIndex, limit }));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call receiveProjectsByClient', () => {
      store.dispatch(actionsProjects.receiveProjectsByClient({
        projects, clientId, startIndex, limit,
      }));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call sidebarSelectProject', () => {
      store.dispatch(actionsProjects.sidebarSelectProject({ projectId }));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call collapsibleSidebarSelect', () => {
      store.dispatch(actionsProjects.collapsibleSidebarSelect(true));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call projectListLoadedSuccess', () => {
      store.dispatch(actionsProjects.projectListLoadedSuccess([]));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call addDatasetToProjectSuccess', () => {
      store.dispatch(actionsProjects.addDatasetToProjectSuccess('Dataset added', { id: '123' }, projectId));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call fetchProject', () => {
      store.dispatch(actionsProjects.fetchProject(projectId));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call refreshProjectsByClient', () => {
      store.dispatch(actionsProjects.refreshProjectsByClient(clientId));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call fetchProjectsByClient', () => {
      store.dispatch(actionsProjects.fetchProjectsByClient({ clientId, startIndex, limit }));
      expect(store.getActions()).toMatchSnapshot();
    });

    test('should call fetchAllProjects', () => {
      store.dispatch(actionsProjects.fetchAllProjects());
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
