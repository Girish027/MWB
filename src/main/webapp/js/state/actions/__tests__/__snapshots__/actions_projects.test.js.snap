// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions_projects <Function call /> should call addDatasetToProjectSuccess 1`] = `
Array [
  Object {
    "dataset": Object {
      "id": "123",
    },
    "projectId": "123",
    "serverMessage": "Dataset added",
    "type": "ADD_DATASET_TO_PROJECT_SUCCESS",
  },
]
`;

exports[`actions_projects <Function call /> should call collapsibleSidebarSelect 1`] = `
Array [
  Object {
    "isCollapsibleModelOpen": true,
    "type": "COLLAPSIBLE_SIDEBAR_SELECT",
  },
]
`;

exports[`actions_projects <Function call /> should call fetchAllProjects 1`] = `
Array [
  Object {
    "type": "REQUEST_PROJECTS_ALL",
  },
]
`;

exports[`actions_projects <Function call /> should call fetchProject 1`] = `
Array [
  Object {
    "projectId": "123",
    "type": "REQUEST_PROJECT",
  },
]
`;

exports[`actions_projects <Function call /> should call fetchProjectsByClient 1`] = `
Array [
  Object {
    "clientId": "14",
    "limit": 100,
    "startIndex": 0,
    "type": "REQUEST_PROJECTS_BY_CLIENT",
  },
]
`;

exports[`actions_projects <Function call /> should call projectListLoadedSuccess 1`] = `
Array [
  Object {
    "newProjectList": Array [],
    "type": "PROJECT_LIST_LOADED_SUCCESS",
  },
]
`;

exports[`actions_projects <Function call /> should call receiveProjectsByClient 1`] = `
Array [
  Object {
    "clientId": "14",
    "limit": 100,
    "projects": Array [],
    "startIndex": 0,
    "type": "RECEIVE_PROJECTS_BY_CLIENT",
  },
]
`;

exports[`actions_projects <Function call /> should call refreshProjectsByClient 1`] = `
Array [
  Object {
    "clientId": "14",
    "limit": 500,
    "startIndex": 0,
    "type": "REQUEST_PROJECTS_BY_CLIENT",
  },
]
`;

exports[`actions_projects <Function call /> should call requestAllProjects 1`] = `
Array [
  Object {
    "type": "REQUEST_PROJECTS_ALL",
  },
]
`;

exports[`actions_projects <Function call /> should call requestProject 1`] = `
Array [
  Object {
    "projectId": "123",
    "type": "REQUEST_PROJECT",
  },
]
`;

exports[`actions_projects <Function call /> should call requestProjectsByClient 1`] = `
Array [
  Object {
    "clientId": "14",
    "limit": 100,
    "startIndex": 0,
    "type": "REQUEST_PROJECTS_BY_CLIENT",
  },
]
`;

exports[`actions_projects <Function call /> should call sidebarSelectProject 1`] = `
Array [
  Object {
    "projectId": "123",
    "type": "SIDEBAR_SELECT_PROJECT",
  },
]
`;

exports[`actions_projects <Function call /> should call toggleSidebarVisible 1`] = `
Array [
  Object {
    "type": "PROJECT_LIST_TOGGLE_VISIBLE",
    "visible": true,
  },
]
`;

exports[`actions_projects createProject: should do post call with required data 1`] = `"/nltools/private/v1/clients/14/projects"`;

exports[`actions_projects deleteProjectById: should do delete call with required data 1`] = `"/nltools/private/v1/clients/14/projects/123"`;

exports[`actions_projects demoteProjectById: should do put call with projectId 1`] = `"/nltools/private/v1/clients/14/projects/123/demote"`;

exports[`actions_projects onCreateProjectFailure: should dispatch actions to display bad request message when project creation failed 1`] = `
Array [
  Object {
    "modalDialogState": null,
    "type": "MODAL_DIALOG_CHANGE",
  },
  Object {
    "type": "BAD_REQUEST",
    "value": "Unable to create project",
  },
]
`;

exports[`actions_projects onCreateProjectFailure: should dispatch actions to display warning message when project creation failed 1`] = `
Array [
  Object {
    "modalDialogState": null,
    "type": "MODAL_DIALOG_CHANGE",
  },
  Object {
    "type": "WARNING_REQUEST",
    "value": "Unable to create project",
  },
]
`;

exports[`actions_projects onCreateProjectFailure: should dispatch actions to display warning message when project creation failed without data 1`] = `
Array [
  Object {
    "modalDialogState": null,
    "type": "MODAL_DIALOG_CHANGE",
  },
  Object {
    "type": "BAD_REQUEST",
    "value": undefined,
  },
]
`;

exports[`actions_projects onCreateProjectSuccess: should dispatch actions to create project and display good request message on project create success 1`] = `
Array [
  Object {
    "project": Object {
      "name": "TestProject",
    },
    "type": "ADD_NEW_PROJECT",
  },
  Object {
    "modalDialogState": Object {
      "dispatch": [Function],
      "header": "Upload Dataset",
      "history": Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/datasets?clientid=&appid=&projectid=undefined",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
      "project": Object {
        "name": "TestProject",
      },
      "projectId": undefined,
      "type": "DatasetDialog",
    },
    "type": "MODAL_DIALOG_CHANGE",
  },
  Object {
    "type": "GOOD_REQUEST",
    "value": "Model TestProject created.",
  },
]
`;

exports[`actions_projects onCreateProjectSuccess: should dispatch actions to create project and display good request message on project create success without data 1`] = `
Array [
  Object {
    "project": Object {},
    "type": "ADD_NEW_PROJECT",
  },
  Object {
    "modalDialogState": Object {
      "dispatch": [Function],
      "header": "Upload Dataset",
      "history": Object {
        "push": [MockFunction] {
          "calls": Array [
            Array [
              "/datasets?clientid=&appid=&projectid=undefined",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
      "project": Object {},
      "projectId": undefined,
      "type": "DatasetDialog",
    },
    "type": "MODAL_DIALOG_CHANGE",
  },
  Object {
    "type": "GOOD_REQUEST",
    "value": "Model undefined created.",
  },
]
`;

exports[`actions_projects onDeleteProjectSuccess: should dispatch actions to delete project and display good request message on project delete success 1`] = `
Array [
  Object {
    "id": "123",
    "type": "PROJECT_DELETE",
  },
  Object {
    "type": "GOOD_REQUEST",
    "value": "Model  deleted.",
  },
]
`;

exports[`actions_projects onMarkDeployableModelFailure: should dispatch actions to set failed status and display bad request message when mark for deploy operation failed 1`] = `
Array [
  Object {
    "type": "BAD_REQUEST",
    "value": "Unable to mark model version 12 as deployable.",
  },
]
`;

exports[`actions_projects onMarkDeployableModelFailure: should dispatch actions to set failed status and display bad request message when mark for deploy operation failed without data 1`] = `
Array [
  Object {
    "type": "BAD_REQUEST",
    "value": "Unable to mark model version 12 as deployable.",
  },
]
`;

exports[`actions_projects onProjectFailure: should dispatch actions to set failed status and display bad request message when project operation failed 1`] = `
Array [
  Object {
    "projectId": "123",
    "status": "FAILED",
    "type": "PROJECT_STATUS_UPDATE",
  },
  Object {
    "type": "BAD_REQUEST",
    "value": "Unable to perform project action",
  },
]
`;

exports[`actions_projects onProjectFailure: should dispatch actions to set failed status and display bad request message when project operation failed without data available 1`] = `
Array [
  Object {
    "projectId": "123",
    "status": "FAILED",
    "type": "PROJECT_STATUS_UPDATE",
  },
  Object {
    "type": "BAD_REQUEST",
    "value": undefined,
  },
]
`;

exports[`actions_projects onProjectFailure: should dispatch actions to set warning status and display bad request message when project operation failed 1`] = `
Array [
  Object {
    "projectId": "123",
    "status": "FAILED",
    "type": "PROJECT_STATUS_UPDATE",
  },
  Object {
    "type": "WARNING_REQUEST",
    "value": "Unable to perform project action",
  },
]
`;

exports[`actions_projects onPromoteProjectSuccess: should dispatch actions to promote project to global model and demote already existing project to node level model 1`] = `
Array [
  Object {
    "projects": Array [
      Object {
        "id": "123",
        "name": "TestProject",
        "type": "NODE",
      },
      Object {
        "id": "234",
        "name": "Root_Intent",
        "type": "GLOBAL",
      },
    ],
    "type": "PROJECT_UPDATE",
  },
  Object {
    "projectId": "123",
    "status": "COMPLETED",
    "type": "PROJECT_STATUS_UPDATE",
  },
  Object {
    "type": "GOOD_REQUEST",
    "value": "Model promoted to global level.",
  },
]
`;

exports[`actions_projects onPromoteProjectSuccess: should dispatch actions to promote project to global model and demote already existing project to node level model without data available 1`] = `
Array [
  Object {
    "projects": Array [],
    "type": "PROJECT_UPDATE",
  },
  Object {
    "projectId": "123",
    "status": "COMPLETED",
    "type": "PROJECT_STATUS_UPDATE",
  },
  Object {
    "type": "GOOD_REQUEST",
    "value": "Model promoted to global level.",
  },
]
`;

exports[`actions_projects onUnmarkDeployableModelFailure: should dispatch actions to set failed status and display bad request message when unmark operation failed 1`] = `
Array [
  Object {
    "type": "BAD_REQUEST",
    "value": "Unable to unmark model version 12 from deployable.",
  },
]
`;

exports[`actions_projects onUnmarkDeployableModelFailure: should dispatch actions to set failed status and display bad request message when unmark operation failed without data 1`] = `
Array [
  Object {
    "type": "BAD_REQUEST",
    "value": "Unable to unmark model version 12 from deployable.",
  },
]
`;

exports[`actions_projects onUpdateProjectSuccess: should dispatch actions to update project status and display good request message on project update success 1`] = `
Array [
  Object {
    "projects": Array [
      Object {
        "0": Object {
          "id": "123",
          "name": "TestProject",
        },
      },
    ],
    "type": "PROJECT_UPDATE",
  },
  Object {
    "projectId": "123",
    "status": "COMPLETED",
    "type": "PROJECT_STATUS_UPDATE",
  },
  Object {
    "type": "GOOD_REQUEST",
    "value": "Model updated.",
  },
]
`;

exports[`actions_projects onUpdateProjectSuccess: should dispatch actions to update project status and display good request message on project update success without response available 1`] = `
Array [
  Object {
    "projects": Array [
      Object {},
    ],
    "type": "PROJECT_UPDATE",
  },
  Object {
    "projectId": "123",
    "status": "COMPLETED",
    "type": "PROJECT_STATUS_UPDATE",
  },
  Object {
    "type": "GOOD_REQUEST",
    "value": "Model updated.",
  },
]
`;

exports[`actions_projects promoteProjectById: should do put call with required data 1`] = `"/nltools/private/v1/clients/14/projects/234/promote?globalProjectName=Root_Intent&globalProjectId=123"`;

exports[`actions_projects updateByProject: should do patch call to update project without data 1`] = `Array []`;

exports[`actions_projects updateByProject: should do post call with required data 1`] = `"/nltools/private/v1/clients/14/projects/123"`;
